tests/tenancy_test.go

```diff
@@ -0,0 +1,150 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/mesosphere/kudo-spark-operator/tests/utils"
+	log "github.com/sirupsen/logrus"
+	"gotest.tools/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+)
+
+func TestTenancyTwoOperatorsDifferentNamespaces(t *testing.T) {
+	operators := operatorBuilder(2, true, true)
+	for _, operator := range operators {
+		err := operator.InstallSparkOperator(true)
+		assert.NilError(t, err)
+	}
+
+	verifyComponents(t, operators)
+	verifyWorkloads(t, operators)
+
+	// verify CRDs are present after one of the operators is deleted
+	operators[0].CleanUp()
+	assert.Assert(t, crdsInstalled(t), "CRDs are not present!")
+
+	// check that CRDs are deleted after no operator instances left
+	operators[1].CleanUp()
+	assert.Assert(t, !crdsInstalled(t), "CRDs are not deleted!")
+}
+
+func TestTenancyTwoOperatorsSingleNamespace(t *testing.T) {
+	operators := operatorBuilder(2, false, true)
+	for _, operator := range operators {
+		err := operator.InstallSparkOperator(false)
+		assert.NilError(t, err)
+		defer operator.CleanUp()
+	}
+
+	verifyComponents(t, operators)
+	verifyWorkloads(t, operators)
+}
+
+func TestTenancyTwoOperatorsSameNameDifferentNamespaces(t *testing.T) {
+	operators := operatorBuilder(2, true, false)
+	for _, operator := range operators {
+		err := operator.InstallSparkOperator(true)
+		assert.NilError(t, err)
+	}
+
+	verifyComponents(t, operators)
+	verifyWorkloads(t, operators)
+
+}
+
+func verifyComponents(t *testing.T, operators []*utils.SparkOperatorInstallation) {
+	serviceAccounts := []string{"spark-operator-service-account", "spark-service-account"}
+	services := []string{"spark-webhook", "spark-operator-metrics"}
+	roles := []string{"spark-driver-role", "%s-spark-role"}
+
+	for _, operator := range operators {
+		for _, service := range services {
```

Comment:
I'd suggest wrapping each following subtests (verify Services, ServiceAccounts, Roles etc.) into `t.Run`, so making them atomical. It will more comply with the unit testing philosophy.

The example of how it could looks like
```suggestion
for _, operator := range operators {
        t.Run("TestServices", func(t *testing.T) {
		for _, service := range services {
					serviceName := fmt.Sprint(operator.InstanceName, "-", service)
 			log.Infof("Checking Service \"%s\" is created in namespace \"%s\" for \"%s\"", serviceName,
 				operator.Namespace, operator.InstanceName)
 			result, err := operator.K8sClients.CoreV1().Services(operator.Namespace).Get(fmt.Sprint(serviceName), v1.GetOptions{})
 			assert.NilError(t, err)
 			assert.Equal(t, result.Labels["kudo.dev/instance"], operator.InstanceName)
 		}
 	}(t)

```

Reply:
Also, the tests which are wrapped into `t.Run` could be run in parallel. The optimization can bring faster execution and atomical the tests fails.